### PR TITLE
delete the favcache and update the favoriting engines / test

### PR DESCRIPTION
**Pull request by oconnor663**
@maxtaco @patrickxb 

Client side changes for https://github.com/keybase/keybase/pull/536 and https://keybase.atlassian.net/browse/CORE-1535.

I've deleted the favcache entirely, so the three favorites engines always hit the network. I figured we'd rather do the simplest correct thing for now and worry about performance down the line. Let me know if you guys feel differently. If we keep the cache, we'll have to decide how frequently a running service should purge it.

The APIs here, including the `keybase favorite *` commands, reflect the fact that right now we're storing only the TLFID and not trying to represent the canonical name on the MySQL side yet. That makes the commands pretty useless except for testing, but they're only exposed through `commands_devel.go`, so users won't see them.

### CODE REVIEW

**Comment by oconnor663** [comment]
@strib just saw your comment on https://github.com/keybase/keybase/pull/536; sounds like I'll need to rework this.

**Comment by strib** [comment]
Sorry, I didn't know you'd be updating the protocol that was already in place!  Let me know if it's a problem, but I do think using the name is the right thing for the favorites list.

**Comment by oconnor663** [comment]
I got pretty aggressive with AVDL changes, and I can totally put things back :)

There are some parts I'm still interested in changing, though. For example, the `private` field; it seems weird to me to have the client store a copy of that state. How do we feel about the MD server and the MySQL db keeping their own copies of the same metadata?

**Comment by maxtaco** [comment]
Without a cache, how much traffic will there be to the server? In other
parts of kbfs performance is justatrocious  without local caching.

**Comment by patrickxb** [comment]
I'm majorly confused why the folder name isn't being used...

**Comment by oconnor663** [comment]
@strib @maxtaco @patrickxb ok this is ready for review using the original RPCs.

File: go/engine/favorite_delete.go [hunk 1]
```diff
@@ -49,6 +51,13 @@ func (e *FavoriteDelete) Run(ctx *Context) error {
 	if e.arg == nil {
 		return fmt.Errorf("FavoriteDelete arg is nil")
 	}
-	e.G().FavoriteCache.Delete(e.arg.Folder)
-	return nil
+	_, err := e.G().API.Post(libkb.APIArg{
+		Endpoint:    "kbfs/favorite/delete",
+		NeedSession: true,
+		Args: libkb.HTTPArgs{
+			"tlf_name": libkb.S{Val: string(e.arg.Folder.Name)},
```
Comment (patrickxb) [inline go/engine/favorite_delete.go]: Folder.Name is already a string, right?

File: go/engine/favorite_list.go [hunk 2]
```diff
@@ -40,12 +43,29 @@ func (e *FavoriteList) SubConsumers() []libkb.UIConsumer {
...
 func (e *FavoriteList) Run(ctx *Context) error {
+	res, err := e.G().API.Get(libkb.APIArg{
+		Endpoint:    "kbfs/favorite/list",
+		NeedSession: true,
+		Args:        libkb.HTTPArgs{},
+	})
+	if err != nil {
+		return err
+	}
+	err = res.Body.UnmarshalAgain(&e.result)
```
Comment (patrickxb) [inline go/engine/favorite_list.go]: could you use API.GetDecode() instead of UnmarshalAgain?

File: go/engine/favorite_delete.go (hunk 1 above)
Comment (oconnor663) [inline go/engine/favorite_delete.go]: Yep. Paste-o from the previous version of this diff.

**Comment by patrickxb** [comment]
LGTM, minor questions.

**Comment by oconnor663** [comment]
Addressed @patrickxb's comments. I don't expect CI tests to pass until I land https://github.com/keybase/keybase/pull/536 though.

**Pull request merged**